numbast/src/numbast/tools/cuda_oxide_binding_generator.py

```
@@ -0,0 +1,721 @@
+                )
+            rust_constant_name = rust_identifier(name)
+            item = (enum_type, value)
+            previous = occupied.get(rust_constant_name)
+            if previous is not None:
+                previous_name, previous_item = previous
+                if previous_name == name and previous_item == item:
+                    continue
+                raise BindingModelError(
+                    [
+                        f"constant name collision: {previous_name!r} and {name!r} "
+                        f"both map to {rust_constant_name!r}"
```

Ruff (0.16.3)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)